File: frontend/src/app/shared/model/api/revision-dto.ts
import {UserId} from '../id-types';

export interface RevisionDto {
  date: number;
  user: UserId;
  revisionType: RevisionType;
  revisionValueType: RevisionValueType;
  changedField: string;
  oldValue: any;
  newValue: any;
}
type RevisionType = 'ADD' | 'MOD' | 'DEL';
type RevisionValueType = 'STRING' | 'USER_COLLECTION';

File: backend/burning-okr/burning-okr-app/src/main/java/org/burningokr/dto/okr/RevisionDto.java
package org.burningokr.dto.okr;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.burningokr.model.okr.RevisionValueType;
import org.hibernate.envers.RevisionType;

import java.util.Date;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RevisionDto {

  private Date date;

  private UUID user;

  private RevisionType revisionType;

  private String changedField;

  private RevisionValueType revisionValueType;

  private Object oldValue;

  private Object newValue;
}
